drivers/frequency/adf5902/adf5902.c

```
@@ -0,0 +1,1046 @@
+		return FAILURE;
+
+	dev->tx_ramp_clk = init_param->tx_ramp_clk;
+
+	if ((init_param->tx_data_invert != AD5902_TX_DATA_INV_DISABLE)
+	    && (init_param->tx_data_invert != AD5902_TX_DATA_INV_ENABLE))
+		return FAILURE;
+
+	dev->tx_data_invert = init_param->tx_data_invert;
+
+	if (init_param->clk2_div_no > (ADF5902_CLK_DIV_SEL_3 + 1))
+		return FAILURE;
+
+	dev->clk2_div_no = init_param->clk2_div_no;
+
+	for (i = 0; i < dev->clk2_div_no; i++)
+		if(init_param->clk2_div[i] > ADF5902_MAX_CLK_DIV_2)
+			return FAILURE;
+
+	dev->clk2_div = init_param->clk2_div;
+
+	if((init_param->clk_div_mode != ADF5902_CLK_DIV_OFF) &&
+	    (init_param->clk_div_mode != ADF5902_FREQ_MEASURE) &&
+	    (init_param->clk_div_mode != ADF5902_RAMP_DIV))
+		return FAILURE;
+
+	dev->clk_div_mode = init_param->clk_div_mode;
+
+	if((init_param->le_sel != ADF5902_LE_FROM_PIN) &&
+	    (init_param->le_sel != ADF5902_LE_SYNC_REFIN))
+		return FAILURE;
+
+	dev->le_sel = init_param->le_sel;
+
+	if (init_param->cp_current > ADF5902_CP_CURRENT_4MA48)
+		return FAILURE;
+
+	dev->cp_current = init_param->cp_current;
+
+	if ((init_param->cp_tristate_en != ADF5902_CP_TRISTATE_DISABLE)
+	    && (init_param->cp_tristate_en != ADF5902_CP_TRISTATE_ENABLE))
+		return FAILURE;
+
+	dev->cp_tristate_en = init_param->cp_tristate_en;
+
+	if(init_param->ramp_status != ADF5902_TEST_BUS_NONE &&
+	    init_param->ramp_status != ADF5902_RAMP_COMPL_TO_MUXOUT &&
+	    init_param->ramp_status != ADF5902_RAMP_DOWN_TO_MUXOUT &&
+	    init_param->ramp_status != ADF5902_TEMP_SENS_TO_ATEST &&
+	    init_param->ramp_status != ADF5902_TEMP_SENS_TO_ADC)
+		return FAILURE;
+
+	dev->ramp_status = init_param->ramp_status;
+
+	if(init_param->ramp_mode != ADF5902_CONT_SAWTOOTH &&
+	    init_param->ramp_mode != ADF5902_SAWTOOTH_BURST &&
+	    init_param->ramp_mode != ADF5902_CONTINUOUS_TRIANGULAR &&
+	    init_param->ramp_mode != ADF5902_SINGLE_RAMP_BURST)
+		return FAILURE;
```

Nitpick: Instead of returning just FAILURE, I would return a different value for each if
Like -0x100, -0x200, .. (leaving the first byte empty in order to not be confused with codes from ernno.h)
This will be useful for debuging.
Or at least -EINVAL for all :)

---

added `return -EINVAL` for all init_param checks.